yorkie/src/main/kotlin/dev/yorkie/core/Attachment.kt

```diff
@@ -4,7 +4,7 @@ import dev.yorkie.document.Document
 
 internal data class Attachment(
     val document: Document,
-    val isRealTimeSync: Boolean,
+    var isRealTimeSync: Boolean,
```

Comment:
having mutable properties for a state could be dangerous, as value changes in such properties do not emit changes to subscirbers, and `isRealTimeSync` is used in a `Flow.map`.
we should keep it immutable and always create a new instance of an Attachment when there are changes.
this is why I changed peerPresences to a immutable map in previous work.

`remoteChangeEventReceived` is a special case, since it is only used inside a while loop, and changes do not need emissions.  but it would be better to make it immutable.

Reply:
thank you for explaining it in detail.
I made `isRealTimeSync` and `remoteChangeEventReceived` immutable: [56aa510](https://github.com/yorkie-team/yorkie-android-sdk/pull/90/commits/56aa51009e6a65328f9a84cf5876b28b53897868)